src/System.Console/tests/SyncTextReader.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using Xunit;
+
+public class SyncTextReaderTests
+{
+    // NOTE: These tests test the underlying SyncTextReader by
+    // accessing the Console.In TextReader (which in fact is a SyncTextReader).
+
+    static readonly string[] s_testLines = new string[] {
+        "3232 Hello32 Hello 5032 Hello 50 532 Hello 50 5 aTrueaabcdbc1.23123.4561.23439505050System.ObjectHello World",
+        "32",
+        "",
+        "32 Hello",
+        "",
+        "32 Hello 50",
+        "",
+        "32 Hello 50 5",
+        "",
+        "32 Hello 50 5 a",
+        "",
+        "True",
+        "a",
+        "abcd",
+        "bc",
+        "1.23",
+        "123.456",
+        "1.234",
+        "39",
+        "50",
+        "50",
+        "50",
+        "System.Object",
+        "Hello World",
+    };
+
+    private static void Test(string content, Action<StreamReader> action)
+    {
+        TextWriter savedStandardOutput = Console.Out;
+        TextReader savedStandardInput = Console.In;
+
+        try
+        {
+            using (MemoryStream memStream = new MemoryStream())
+            {
+                using (StreamWriter sw = new StreamWriter(memStream))
+                {
+                    sw.Write(content);
+                    sw.Flush();
```

Comment:
Code stolen from ReadAndReadLines test. Didn't want to change the existing
style.
On Mar 10, 2015 4:31 PM, "Stephen Toub" <notifications@github.com> wrote:

> In src/System.Console/tests/SyncTextReader.cs
> <https://github.com/dotnet/corefx/pull/1114#discussion_r26132003>:
>
> > +        "Hello World",
> > +    };
> > +
> > +    private static void Test(string content, Action<StreamReader> action)
> > +    {
> > +        TextWriter savedStandardOutput = Console.Out;
> > +        TextReader savedStandardInput = Console.In;
> > +
> > +        try
> > +        {
> > +            using (MemoryStream memStream = new MemoryStream())
> > +            {
> > +                using (StreamWriter sw = new StreamWriter(memStream))
> > +                {
> > +                    sw.Write(content);
> > +                    sw.Flush();
>
> You could actually end the using block here, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dotnet/corefx/pull/1114/files#r26132003>.
>